### DISCUSSION TITLE
Defining "metadata enrichment into sample interface"

### DISCUSSION THREAD

**Discussion by mslarae13** [microbiomedata/issues#765] [General]
Relates to milestone 3.3:
- https://github.com/microbiomedata/issues/issues/473

Any metadata enrichment is a success. As a "now, next, later" goal for this mile stone, I propose:

**NOW**
- Evaluate proof-of-concept metadata enrichment routine from the pilot for adoption to NMDC submission
    - Can we just add a "Enrich my metadata using geo_loc_tools check box?
      - On the back end, add the metadata
      - WAY less UI updates.
      - Who makes sure this gets run?
      - Who does the 'enrichment'   

**Next** - FY25?
-   Direct integration with the submission portal?
      - UI development
      - UI integration
      - Less manual & automatic
      - Who?
- User enters a lat_lon and the front end auto-populates this additional metadata information

**Later** - Renewal?
- Expand the fields and metadata that can be autopopulated
  - what tools exist?
  - Where do we get the information?

**Comment by mslarae13** [comment]
@emileyfadrosh @cmungall @lamccue @aclum @ssarrafan

**Comment by aclum** [comment]
For the 'now' implementation I worry about having submissions where a user hasn't approved of the value for the geographic location name. I'd like to look into skipping the now implementation in favor of starting on the 'next' section.

**Reply by mslarae13** [reply]
@aclum what do you mean "hasn't approved of the value for the geographic location name"? The geo location name would come from their submission. So the name should be accurate from their addition.

I do not think we can accomplish the next by the time this milestone is due, so we'd need to postpone this milestone.

**Reply by aclum** [reply]
As written it seemed to me like we'd only populate geographic location name after the submission portal submission was complete in which case the user wouldn't have a chance to review it. Let me know if i read that incorrectly.

**Comment by lamccue** [comment]
It seems to me that this Milestone (3.3) as it was written is obsolete. 

The text specifically describes taking the [sample-annotator](https://github.com/microbiomedata/sample-annotator/tree/main) script, and making it part of the Submission Portal. That script hasn't been touched in a couple years. And it is designed to "tidy and infer missing data according to a specified schema (currently MIxS)". But our schema is _way_ beyond MIxS, and the validation in the Submission Portal handles much of the "tidying" that the script does. At least that is how it appears to me.

There is additional text that goes on to suggest more improvements to the Submission Portal, e.g., using the geolocation to suggest ENVO or other environment terms. That is a bigger task, and a topic that has come up in the metadata meetings (and see #metadata Slack channel). If we want to pursue this, it should probably be broken out into a more specific milestone with a realistic due date.
